### ISSUE TITLE
[SQLInstance] Locked with unknown field "status.observedGeneration"

### ISSUE THREAD

**Issue by b4nst** [GoogleCloudPlatform/k8s-config-connector#467]
## Checklist

- [ X ] I did not find a related open issue.
- [ X ] I did not find a solution in the troubleshooting guide: (https://cloud.google.com/config-connector/docs/troubleshooting)
- [ X ] If this issue is time-sensitive, I have submitted a corresponding issue with [GCP support](https://cloud.google.com/support-hub).

## Bug Description

My SQLInstance (it seems to apply also to some IAM resources and SQLDatabase) is stuck. Looking at KCC control manager logs I can see a bunch of:
```
admission webhook \"deny-unknown-fields.cnrm.cloud.google.com\" denied the request: unknown field \"status.observedGeneration\"
```
which I can also see on `deletiondefender` when trying to remove the resource.

I've tried as last resort to remove the finalizer to be able to force remove the resource (abandon is already set) but when trying to edit the resource I also get the same error:
```
error: sqlinstances.sql.cnrm.cloud.google.com "<db-name>" could not be patched: admission webhook "deny-unknown-fields.cnrm.cloud.google.com" denied the request: unknown field "status.observedGeneration"
```

Is there something else I can try to remove this resource?

### Kubernetes Cluster Version

Server Version: v1.18.17-gke.1900

### Config Connector Version

1.45.0

### Config Connector Mode

cluster⏎

## Log Output

```
{"severity":"error","logger":"controller-runtime.manager.controller.sqldatabase-controller","msg":"Reconciler error","reconciler group":"sql.cnrm.cloud.google.com","reconciler kind":"SQLDatabase","name":"<dbname>","namespace":"<ns>","error":"error with update call to API server: admission webhook \"deny-unknown-fields.cnrm.cloud.google.com\" denied the request: unknown field \"status.observedGeneration\""}
```

**Comment by maqiuyujoyce** [comment]
Hi @b4nst , sorry that you've ran into this issue.

`status.observedGeneration` is a field Config Connector introduced in [v1.46.0](https://github.com/GoogleCloudPlatform/k8s-config-connector/releases/tag/v1.46.0) and it is not expected to be an issue in v1.45.0. Could you confirm all the pods that Config Connector is using are at version 1.45.0? (E.g. the version of cnrm-controller-manager-0 pod can be found by running `kubectl get pod cnrm-controller-manager-0 -n cnrm-system -o jsonpath='{.metadata.annotations.cnrm\.cloud\.google\.com/version}'`). I wondered if you happened to have upgraded Config Connector partially.

We'd like to reproduce the issue before we can suggest a solid workaround. Could you provide more details about how you run into this issue, and how it can reproduced step by step (if possible)? And it would be great if you can share the yaml you used to create the resources.

Meanwhile, you can probably try editing the resource and removing the `status.observedGeneration` field together with the finalizer. Without being able to reproduce this issue, I can't verify if this works, but may worth trying out.

**Comment by b4nst** [comment]
Hi @maqiuyujoyce thanks for your help!

It happened after a gke cluster update from v1.17.x to v1.18.x so I suspect you're right about a KCC partial update. In the meantime since we needed the cluster operational, I destroyed it and recreated it. Everything is fixed, there is no issues left on KCC and it acquires back the existing resources. 

I will close this issue for now, and will try to reproduce it on a gke update of another cluster as soon as I have some time.

P.S.: I tried to remove both finalizers and `status.observedGeneration` in the same edit but it did not work (same error).
